overlord/snapstate/snapstate.go

```diff
@@ -613,16 +613,26 @@ func removeInactiveRevision(st *state.State, name string, revision snap.Revision
 }
 
 // canRemove verifies that a snap can be removed.
-func canRemove(si *snap.Info, active bool) bool {
+func canRemove(si *snap.Info, snapst *SnapState, removeAll bool) bool {
+	// removing single revisions is generally allowed
+	if !removeAll {
```

Comment:
Is this flag set when it's the last revision that is being removed?  We need to be careful to not allow anything at all to be removed because a specific revision was provided, and that revision is actually the last one.